src/ui/services/networkService.ts

```diff
@@ -23,50 +23,72 @@ export class NetworkService extends EventEmitter {
     currentConnection: SocketIOClient.Socket;
 
+    /** List of all users. */
+    users: user.User[];
+
+    me: user.User;
+
+    imHost: boolean;
+
     constructor() {
         super();
 
         this.lobbies = [];
+        this.users = [];
     }
 
-    connectToCentral(url: string): Promise<void> {
+    connectToCentral(url: string, username = "Unknown", iconId: number): Promise<void> {
         return new Promise<void>((resolve, reject) => {
             this.currentConnection = io.connect(url, { reconnection: false });
             this.currentConnection.on("connect", resolve);
-
-            const [add, update, remove] = this.buildListUpdater("lobbylist", this.lobbies, {
+            var userInfo = {
+                username: username,
+                iconId: iconId
+            };
+            this.me = {id: -1, username: username, iconId: iconId};
+            this.currentConnection.emit("user.userInfo", userInfo);
+            this.currentConnection.on("user.userInfo", (c) => {
+                this.me = c;
+            });
+            
+            const [addLobby, updateLobby, removeLobby] = this.buildListUpdater("lobbylist", this.lobbies, {
+                id: "id",
                 name: "name",
-                creator: "creator",
+                owner: "owner",
                 playerLimit: "playerLimit",
                 playerCount: "playerCount",
-                gamemodeName: "gameMode",
+                gamemodeName: "gamemodeName",
                 hasPassword: "requirePassword",
                 address: "address",
                 port: "port"
             });
-            this.currentConnection.on("lobbylist-add", add);
-            this.currentConnection.on("lobbylist-update", update);
-            this.currentConnection.on("lobbylist-remove", remove);
+            this.currentConnection.on("lobbylist-add", addLobby);
+            this.currentConnection.on("lobbylist-update", updateLobby);
+            this.currentConnection.on("lobbylist-remove", removeLobby);
+
+            const [addUser, updateUser, removeUser] = this.buildListUpdater("users", this.users, {
+                id: "id",
+                username: "username",
+                idIcon: "idIcon"
+            });
+            this.currentConnection.on("users-add", addUser);
+            this.currentConnection.on("users-update", updateUser);
+            this.currentConnection.on("users-remove", removeUser);
 
             //Sending the options to the server
-            var options = {
-                name: "name",
-                creator: "creator",
-                playerLimit: "10",
-                gamemodeName: "gameMode",
-                requirePassword: "false",
-                address: "address",
-                port: "port"
-            }
-            this.currentConnection.emit('lobby.create', options);
+            this.currentConnection.on("lobby.list", function(lobbiesList){
```

Comment:
Arrow function, `for (const lobby of lobbiesList)`

Reply:
d